src/aead/aes.rs

```diff
@@ -356,7 +355,7 @@ impl From<Counter> for Iv {
 
 impl Iv {
     pub(super) fn as_bytes_less_safe(&self) -> &[u8; 16] {
-        self.0.as_byte_array()
+        zerocopy::transmute_ref!(&self.0)
```

Comment:
Is there not a convenient way to do it without using macros? Not the end of the world, for sure.

Reply:
The short answer is no; we rely on the fact that `core::mem::transmute` uses compiler magic to ensure that the types' sizes are equal (we also use the same trick to ensure that alignment of the source type is at least as large as the alignment of the destination type). 

The long answer is that it's _technically_ possible, but it has a lot of sharp edges; here are some links if you're curious:
* https://github.com/google/zerocopy/pull/183#discussion_r1208678157
* https://github.com/rust-lang/unsafe-code-guidelines/issues/409
* https://github.com/rust-lang/rust/issues/112090

At some point in the future, we may be able to support this as a function without needing to rely on the const eval hacks. In particular, that'll come once we are able to encode more facts about types in the type system itself. This is ongoing work, but it'll be a while before we're at a place where we can replace `transmute_ref!`.

Reply:
bytemuck can do this without macros. Unfortunately bytemuck doesn't have byteorder types built-in. I wanted to PR them in at some point. So maybe soon.

Reply:
Which bytemuck API are you referring to?

Reply:
Ah looks like you're probably referring to [`cast_ref`](https://docs.rs/bytemuck/latest/bytemuck/fn.cast_ref.html)? Looks like that's not exactly the same API - if the sizes don't match or the source alignment is less than the destination alignment, `cast_ref` panics at runtime, while `transmute_ref!` fails to compile.

Reply:
Yeah I guess that's true.